### DISCUSSION TITLE
Why there is no Float16 type?

### DISCUSSION THREAD

**Discussion by FengJungle** [onnx/onnx-mlir#2998] [Q&A]
As the title, why there is no float16 type? 
![image](https://github.com/user-attachments/assets/33d30cdf-ca29-4b37-a828-3f9836fec892)

**Comment by AlexandreEichenberger** [comment]
We typically add functionality as we encounter it. I would encourage you to contribute this simple extension.

**Reply by FengJungle** [reply]
> We typically add functionality as we encounter it. I would encourage you to contribute this simple extension.

How about use ±65504 as the max/min limit for float16 type?

**Comment by AlexandreEichenberger** [comment]
I am not an expert in the limit values for float16, but if the integer representation you listed above corresponds to min & max float16, feel free to add this case to the switch.
